cognite_toolkit/_cdf_tk/utils/http_client/_client.py

```diff
@@ -0,0 +1,273 @@
+import gzip
```

Comment:
I don't get it, is this going to be used instead of pysdk? why not just set `max_workers=1` to limit concurrency?

Reply:
Multiple reasons.
1. We need to handle items differently than the SDK, see next PR https://github.com/cognitedata/toolkit/pull/1923
2. We support APIendpoints that are not (yet) supported by the SDK, then relying on the internals of the SDK is a risk as earlier risk-review has pointed out.
3. We need to be able to get the json directly, not necessarily use the same dataclass as the SDK.

Note that this is version 2 of an HTTPClient, I made one earlier this summer that I will remove once this has replaced it.

Reply:
1. In the regular SDK, if you have 1000 items and one of them have an issue, all fails. In part to referenced above, we split ItemsRequest to ensure that each item is tried. We will use this in upload and migration commands. 
3. That may be, but changing the PySDK is a slower process than Toolkit and I need this yesterday. 

Note that Toolkit already have its own HTTPClient, this is just a refactoring of that (easier to write a new and replace, than to do an actual refacotring).